### ISSUE TITLE
Accessing internal functions

### ISSUE THREAD

**Issue by Aidajafarbigloo** [KnowledgeCaptureAndDiscovery/somef#626]
Hello @dgarijo ,

We have been exploring ways to enhance the capabilities of our tool, and we believe that utilizing Somef's internal functions could help us in this way. 

Our purpose is to extract and export more comprehensive metadata with the help of Somef. Currently, we are able to obtain the results from Somef in JSON files, but for further processing within our tool, we require access to the extracted metadata from Somef in the form of Python dictionaries.

We tried to modify the Somef source code to get the result from functions like "save_codemeta_output ", which export CodeMeta format of Metadata, in a form of Python dictionary. However, we recognize that accessing the functions, which export metadata, directly would be a more ideal solution, and give us the possibility to use all three main output formats in form of python dictionaries, not stored in JSON files. Could you share any advice on solving this problem?

**Comment by dgarijo** [comment]
Hello,
Thanks for your interest in the tool!
If you use cli_get_data function in the somef_cli class you will obtain a
json file with the full dictionary returned by the tool (no files are
stored). Is this what you are aiming for?

**Comment by Aidajafarbigloo** [comment]
Hello,

Thank you for the answer.
Yes, right. Also we'd like to access the other output formats of Somef (e.g.: CodeMeta). Using cli_get_data function helps to get the main output.

**Comment by dgarijo** [comment]
For ttl you  have to use `data_graph.somef_data_to_graph(repo_data.results)`. This export has to be improved a bit but should work.
For the codemeta dictionary, have a look at `save_codemeta_output`, which will do the transformation from the dictionary to codemeta and print it out. In this case, the function prints the codemeta file rather than return a dictionary. I can separate it in two functions if you prefer

**Comment by Aidajafarbigloo** [comment]
Thanks, it would be great to have them separated and access CodeMeta dictionary and not producing a file. 

I'm trying to test ``save_codemeta_output``, but I face:

<img width="378" alt="image" src="https://github.com/KnowledgeCaptureAndDiscovery/somef/assets/143706305/d3c72851-8a1e-422b-af34-fab415c9cde9">

I've changed my GitHub token, and reconfigured Somef, it shouldn't related to hitting the rate limitation, right?

**Comment by Aidajafarbigloo** [comment]
Thank you, the token was configured correctly, but the error was because of the scope of the token. I was using a token with not sufficient permissions.

**Comment by dgarijo** [comment]
I see. We should refine the errors that we get when the token is incorrect. I guess another issue to open :(

**Comment by dgarijo** [comment]
I think this is mostly done. @juanjemdIos ?

**Comment by juanjemdIos** [comment]
Yes, now we have "somef configure test", which reports the available tokens and provides better logging that shows pending requests and the scope status. These are some of the examples we are currently showing:
Invalid token. Run `somef configure` to set a valid token.
Token lacks required permissions or scopes.
INFO-Remaining repository API requests: 4990 ### Next rate limit reset at: 2026-0.....

**Comment by dgarijo** [comment]
Closing then
